lib/content-patcher/patchers/class-shortcodepullquotepatcher.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Patcher for the [pullquote][/pullquote] elements.
+ *
+ * @package Newspack
+ */
+
+namespace NewspackContentConverter\ContentPatcher\Patchers;
+
+use NewspackContentConverter\ContentPatcher\Patchers\PatcherInterface;
+use NewspackContentConverter\ContentPatcher\Patchers\PatcherAbstract;
+use NewspackContentConverter\ContentPatcher\ElementManipulators\SquareBracketsElementManipulator;
+use NewspackContentConverter\ContentPatcher\ElementManipulators\WpBlockManipulator;
+use NewspackContentConverter\ContentPatcher\ElementManipulators\HtmlElementManipulator;
+
+/**
+ * Patcher class for the [pullquote][/pullquote] elements.
+ *
+ * @package NewspackContentConverter\ContentPatcher\Patchers
+ */
+class ShortcodePullquotePatcher extends PatcherAbstract implements PatcherInterface {
+
+	/**
+	 * SquareBracketsElementManipulator service.
+	 *
+	 * @var SquareBracketsElementManipulator
+	 */
+	private $square_brackets_element_manipulator;
+
+	/**
+	 * WpBlockManipulator service.
+	 *
+	 * @var WpBlockManipulator
+	 */
+	private $wp_block_manipulator;
+
+	/**
+	 * VideoPatcher constructor.
+	 */
+	public function __construct() {
+		$this->square_brackets_element_manipulator = new SquareBracketsElementManipulator();
+		$this->wp_block_manipulator                = new WpBlockManipulator();
+	}
+
+	/**
+	 * See the \NewspackContentConverter\ContentPatcher\Patchers\PatcherInterface::patch_blocks_contents for description.
+	 *
+	 * @param string $source_html   HTML source, original content being converted.
+	 * @param string $source_blocks Block content as result of Gutenberg "conversion to blocks".
+	 *
+	 * @return string|false
+	 */
+	public function patch_blocks_contents( $source_html, $source_blocks ) {
+		$matches_blocks = $this->wp_block_manipulator->match_wp_block( 'wp:shortcode', $source_blocks );
+		if ( ! $matches_blocks ) {
+			return $source_blocks;
+		}
+
+		foreach ( $matches_blocks[0] as $matched_block ) {
+			$block = $matched_block[0];
+			if ( false === strpos( $block, '[/pullquote]' ) ) {
+				continue;
+			}
+
+			$converted_block = $this->convert_shortcode_block_to_pullquote( $block );
+			$source_blocks   = str_replace( $block, $converted_block, $source_blocks );
+		}
+
+		return $source_blocks;
+	}
+
+	/**
+	 * Convert a shortcode block with the pullquote shortcode into a pullquote block.
+	 *
```

Comment:
This is just a suggestion, might not really be important, so please use if you think it's appropriate, or feel free to ignore it otherwise.

A bit more comments might come in handy. Here's what we could add:
```
...
	 *
	 * The pullquote's 'author' attribute gets converted to a <cite> element, e.g.
	 *      <!-- wp:shortcode -->
	 *          [pullquote author"Arthur Author"]pulled text[\pullquote]
	 *      <!-- /wp:shortcode -->
	 * gets converted to
	 *      <!-- wp:pullquote {"align":"left"} -->
	 *          <cite>Arthur Author</cite>
	 *          pulled text
	 *      <!-- /wp:pullquote -->
	 *
...
```

Reply:
Yeah that's a good idea. It will be clearer how the transformation works with better comments. https://github.com/Automattic/newspack-content-converter/pull/18/commits/98de344abf0c5f390a5fa2492b5aa358815f4537